lib/nrf_modem_lib/trace_backends/flash/flash.c

```diff
@@ -0,0 +1,392 @@
+/*
+ * Copyright (c) 2022-2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <sys/errno.h>
+#include <zephyr/drivers/flash.h>
+#include <zephyr/kernel.h>
+#include <zephyr/fs/fcb.h>
+#include <zephyr/storage/flash_map.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/logging/log.h>
+
+#include <modem/trace_backend.h>
+
+LOG_MODULE_REGISTER(modem_trace_backend, CONFIG_MODEM_TRACE_BACKEND_LOG_LEVEL);
+
+#define EXT_FLASH_DEVICE DEVICE_DT_GET(DT_ALIAS(ext_flash))
+#define TRACE_OFFSET FLASH_AREA_OFFSET(modem_trace)
+#define TRACE_SIZE CONFIG_NRF_MODEM_LIB_TRACE_BACKEND_FLASH_PARTITION_SIZE
+
+#define BUF_SIZE CONFIG_NRF_MODEM_LIB_TRACE_BACKEND_FLASH_BUF_SIZE
+
+static trace_backend_processed_cb trace_processed_callback;
+
+static const struct flash_area *modem_trace_area;
+static const struct device *flash_dev;
+static struct flash_sector trace_flash_sectors[CONFIG_NRF_MODEM_LIB_TRACE_FLASH_SECTORS];
+
+struct fcb trace_fcb = {
+#if IS_ENABLED(CONFIG_FCB_ALLOW_FIXED_ENDMARKER)
+	.f_flags = FCB_FLAGS_CRC_DISABLED,
+#endif
+};
+
+/* Store in __noinit RAM to perserve in warm boot. */
+static __noinit uint32_t magic;
+static __noinit size_t stream_buf_written;
+static __noinit size_t trace_bytes_written;
+static __noinit uint8_t stream_buf[BUF_SIZE];
```

Comment:
`magic` is one thing, but do we still need the other three variables to be `__noinit` ?

Reply:
I'll rewrite this